hooks/pre/35-change-evr-extension-ownership.rb

```diff
@@ -0,0 +1,28 @@
+# In Katello 4.16, the 'evr' extension is removed from PostgreSQL and integrated into the Katello database via a migration.
+# This hook ensures the 'evr' extension's ownership is transferred to the 'foreman' user so migrations can act on it.
+
+if (local_foreman_db? || devel_scenario?) && execute("rpm -q postgresql-evr", false, false)
+  if app_value(:noop)
+    logger.debug("Would start postgresql service")
+  else
+    start_services(['postgresql'])
+  end
+
+  database = param_value('foreman', 'db_database') || 'foreman'
+  username = param_value('foreman', 'db_username') || 'foreman'
+
+  if local_db_exists?(database)
+    sql = "UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{username}') WHERE extname='evr'"
+    if app_value(:noop)
+      logger.debug("Would execute: #{sql} to update ownership of the evr extension")
```

Comment:
```suggestion
      logger.debug("Would execute the following SQL statement to update ownership of the evr extension: #{sql}")
```

That way it's more clear where text ends and SQL begins

Reply:
Updated to this message. One more change I hadn't noticed I had locally was updating a couple of log warnings to logger.notice because that seemed to look better in the output.